types/proposal.go

```diff
@@ -35,15 +35,15 @@ type Proposal struct {
 
 // NewProposal returns a new Proposal.
 // If there is no POLRound, polRound should be -1.
-func NewProposal(height int64, round int32, polRound int32, blockID BlockID, daH *DataAvailabilityHeader) *Proposal {
+func NewProposal(height int64, round int32, polRound int32, blockID BlockID) *Proposal {
 	return &Proposal{
 		Type:      tmproto.ProposalType,
 		Height:    height,
 		Round:     round,
 		BlockID:   blockID,
 		POLRound:  polRound,
 		Timestamp: tmtime.Now(),
-		DAHeader:  daH,
+		DAHeader:  blockID.DataAvailabilityHeader,
 	}
 }
```

Comment:
The `Proposal` type now has two references to the data availability header, but that will change when we remove the block id completely.